components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -49,6 +60,35 @@ class AddonManager(
             throw AddonManagerException(throwable)
         }
     }
+
+    /**
+     * Try updates the provided [extensionId] by updating it in the the [store] and re-attaching
+     * all the handlers.
+     */
+    fun tryUpdateExtension(extensionId: String, onFinish: ((Status) -> Unit)) {
```

Comment:
Since we're in the `AddonManager` here, let's not use `Extension` here. It gets a bit confusing. How about just `updateAddon`? And document that if no update is available, nothing will happen? Or if you prefer `maybeUpdateAddon`?

Reply:
👍